client/src/js/components/bhDateEditor.js

```diff
@@ -44,12 +45,15 @@ function bhDateEditorController(bhConstants) {
 
   $ctrl.$onInit = () => {
     $ctrl.label = $ctrl.label || 'FORM.LABELS.DATE';
-
+    $ctrl.allowFutureDate = $ctrl.allowFutureDate || false;
     // options to be passed to datepicker-option
     $ctrl.options = {
       minDate : $ctrl.minDate,
       maxDate : $ctrl.maxDate,
     };
+    if ($ctrl.allowFutureDate) {
```

Comment:
Instead of deleting this, why not just add it in if `$ctrl.allowFutureDate` doesn't exit?  Like this:
```js
$ctrl.options = {
  minDate : $ctrl.minDate,
};

if ($ctrl.allowFutureDate) {
  $ctrl.options.maxDate = $ctrl.maxDate;
}
```